File: core/src/main/java/dungeon/crawler/GameSystem/SaveGame/Serialization/PartyCharacterSave.java
package dungeon.crawler.GameSystem.SaveGame.Serialization;

import dungeon.crawler.GameSystem.Character.Condition;
import dungeon.crawler.GameSystem.Character.Stance;
import dungeon.crawler.GameSystem.Inventory.ItemTypes.EquipmentSlot;

import java.util.ArrayList;
import java.util.Map;

public class PartyCharacterSave {
    public int level;
    public int xp;
    public int strength;
    public int agility;
    public int intelligence;
    public int perception;
    public boolean isHero;
    public int toHit;
    public String charClass;
    /* can we use an enum for the key instead*/
    public Map<EquipmentSlot, String> equipment;

    public int maxHp;
    public int maxMP;
    public int hp;
    public int mp;
    public Stance stance;
    public ArrayList<Condition> conditions;
    public boolean isDead;

    /* will be added to empty Inventory after character initialization */
    public ArrayList<String> inventory;

    /* will be tacked onto charClass MagicSystem */
    public ArrayList<String> spells;

    public PartyCharacterSave(){};


}

File: core/src/main/kotlin/dungeon/crawler/GameSystem/SaveGame/Serialization/PartyCharacterSave.kt
package dungeon.crawler.GameSystem.SaveGame.Serialization

import dungeon.crawler.GameSystem.Character.Condition
import dungeon.crawler.GameSystem.Character.Stance
import kotlinx.serialization.Serializable

@Serializable
data class PartyCharacterSave(
    val name: String,
    val level: Int,
    val xp: Int,
    val strength: Int,
    val agility: Int,
    val intelligence: Int,
    val perception: Int,
    val isHero: Boolean,
    val toHit: Int,
    val charClass: String,
    val equipment: Map<String, String>,
    val maxHp: Int,
    val maxMP: Int,
    val hp: Int,
    val mp: Int,
    val stance: Stance,
    val conditions: ArrayList<Condition>,
    val isDead: Boolean,
    val inventory: ArrayList<String>,
    val spells: ArrayList<String>
)
